src/common/cmdparse.h

```diff
@@ -79,19 +79,16 @@ bool cmd_getval(const cmdmap_t& cmdmap,
 
 // with default
 
-template <typename T>
-bool cmd_getval(
-  const cmdmap_t& cmdmap, std::string_view k,
-  T& val, const T& defval)
+template <typename T, typename V>
+T cmd_getval_or(const cmdmap_t& cmdmap, std::string_view k,
```

Comment:
also, changed the function signature so it's different from the one defined at line 65. otherwise, that one would be chosen instead of this one when the compiler is performing argument-dependent lookup.